cmd/main.go

```diff
@@ -555,3 +566,82 @@ func createControllerManager() (manager.Manager, error) {
 	}
 	return nil, nil
 }
+
+func newAssistedServiceRoute(service *corev1.Service) *routev1.Route {
```

Comment:
Nothing in here looks conditional. Can we create this outside of the app and send in the route name to do the rest of the logic?

Ideally the application wouldn't know about any of this and should just be sent the externally accessible endpoint, but if that's not possible we should get as much of this out of the app as we can.

Reply:
This comment does make me wonder if we could do this work as an `initContainer` instead of app startup. There isn't a way to handle it via OLM packaging mechanisms...so we would either need to make it a pre-req for the user, check out initContainers as a solution, or as part of app startup.

Reply:
> There isn't a way to handle it via OLM packaging mechanisms

Can you elaborate on this? Couldn't this be added the same way we add the service? What's the difference?

Reply:
https://docs.openshift.com/container-platform/4.6/operators/understanding/olm-packaging-format.html

Specifically, under Manifests->Additionally supported types:
> Supported optional object types
Secret
ConfigMap
Service
PodDisruptionBudget
PriorityClass
VerticalPodAutoScaler

Reply:
I just experimented with the init container approach.

`
      - name: init-add-route-and-update-config-map
        image: quay.io/ocpmetal/assisted-service:latest
        command: ['sh', '-c', 'export NAMESPACE=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace) && oc expose service -n $NAMESPACE assisted-service && export ROUTE_URL=$(oc get routes -n $NAMESPACE assisted-service -o jsonpath={.spec.host}) && oc get configmap -n $NAMESPACE assisted-service-config -o yaml | sed -e "s|REPLACE_BASE_URL|http\:\/\/$ROUTE_URL|" | oc apply -f - ']
`

It works fine, creating the route and updating the SERVICE_BASE_URL in the config map. But it runs into the problem where the route remains when the operator is uninstalled.  Reinstalling the operator throws an error in the init container because it attempts to create a route that is already there. I'll need to add more logic to not create a route if it already exists.

Reply:
The init container has replaced the in app code.